asn1tools/source/c/uper.py

```diff
@@ -773,23 +773,41 @@ def format_choice_inner(self, type_, checker):
 
     def format_enumerated_inner(self, type_):
         type_name = self.format_type_name(0, max(type_.root_index_to_data))
-        unique_value = self.add_unique_decode_variable(
+        unique_value = self.add_unique_variable(
             '{} {{}};'.format(type_name),
             'value')
         location = self.location_inner()
 
-        encode_lines = [
-            'encoder_append_non_negative_binary_integer(encoder_p, '
-            'src_p->{}, {});'.format(location,
-                                     type_.root_number_of_bits)
-        ]
+        value_mapping_required = \
+            any([type_.root_data_to_index[data] != type_.root_data_to_value[data]
+                 for data in type_.root_data_to_index.keys()])
```

Comment:
`keys()` is not needed. Iteration is over keys by default.

Reply:
Thanks for the hint, I removed the keys()